src/client/resources/web/js/chat.mjs

```diff
@@ -122,33 +166,76 @@ function connect() {
         /** @type {string} */
         const rawJson = event.data;
         console.log(rawJson);
-		try {
-			const message = parseModServerMessage(rawJson);
-			// For now we only handle chat messages
-			if (message.type === 'chatMessage') {
-				displayChatMessage(message.payload, message.timestamp);
-			}
-		} catch (e) {
-			console.error('Error processing message:', e);
-		}
+        try {
+            const message = parseModServerMessage(rawJson);
+            // For now we only handle chat messages
+            if (message.type === 'chatMessage') {
+                // Skip if we've already seen this message
+                if (displayedMessageIds.has(message.payload.uuid)) {
```

Comment:
If we're deduping it seems we might also want to insert messages in sorted position? This would be possible if we use something like UUIDv7 that is time-orderable and has a hash component.

But I can also understand that might be overkill.

Reply:
Crossed my mind (but using the timestamp) considered it a bit much indeed.